iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -502,6 +502,40 @@ string<Capacity>::append(TruncateToCapacity_t, const T& t) noexcept
     return *this;
 }
 
+template <uint64_t Capacity>
+template <typename T>
+inline constexpr typename std::enable_if<is_char_array<T>::value || is_cxx_string<T>::value, bool>::type
+string<Capacity>::insert(const uint64_t pos, const T& str, const uint64_t count) noexcept
+{
+    if (count > internal::GetSize<T>::call(str))
+    {
+        return false;
+    }
+    const auto new_size = m_rawstringSize + count;
+    if (new_size > Capacity)
+    {
+        return false;
+    }
+    if (Capacity < pos + count)
+    {
+        return false;
+    }
+
+    auto remainder = substr(pos);
+    if (!remainder)
+    {
+        return false;
+    }
+
+    std::memcpy(&(m_rawstring[pos]), internal::GetData<T>::call(str), count);
+    std::memcpy(&(m_rawstring[pos + count]), remainder->c_str(), remainder->size());
```

Comment:
Given the numbers in the warnings m = 18446744073709551615 = max 64 bit integer, B = 9223372036854775807 = m/2  
and reading

https://archive.fosdem.org/2017/schedule/event/valgrind_fortification/attachments/slides/1868/export/events/attachments/valgrind_fortification/slides/1868/valgrind_fortify.html#slide6

https://gcc.gnu.org/onlinedocs/gcc/Object-Size-Checking.html

I have the following theory.

1) gcc tries to determine the  size of the destination buffer, but cannot really bound it (insufficient static information due to pointer arithmetics etc). 
2) It then assumes the bound B by default (see above, whether this is smart or not I cannot say, but in general it is a large number).
3) It also only knows it has to copy count bytes to the buffer and assumes count = m in the worst case.
4) since M > B it issues a warning.

To test this, we could (temporarily) create 
`uint32_t c = (uint32_t) count;` and `uint32_t r = (uint32_t) remainder->size()` and only copy c and r bytes, respectively.

For our tests the casts are ok, but generally they are not. However, it is questionable whether the size (internally `m_rawSize` and return value of `size()` really should be 64 bit integers. I would assume we will not really have strings with >2^32 characters, especially in a static memory setup.

TLDR: the implemented logic is correct from what I can see, but certain compilers fail to detect this due to the way they determine worst case sizes and the type they use to store these sizes.

@FerdinandSpitzschnueffler